command/agent.go

```diff
@@ -368,13 +368,22 @@ func (c *AgentCommand) Run(args []string) int {
 			client.SetNamespace(config.AutoAuth.Method.Namespace)
 		}
 		templateNamespace = client.Headers().Get(consts.NamespaceHeaderName)
+
+		sinkClient, err := client.CloneWithHeaders()
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error cloning client for file sink: %v", err))
+			return 1
+		}
+
+		sinkClient.SetDisableKeepAlives(config.DisableKeepAlivesAutoAuth)
```

Comment:
Do we want the default to be enabling keep-alives, or disabling them? (Since Go favors `false`, here we are enabling them by default I believe.)

Reply:
`config.DisableKeepAlivesAutoAuth` is only true if `auto-auth` is found in the configurable string as seen here: https://github.com/hashicorp/vault/pull/15986/files#diff-f309f8618f6d7bea196a7f4eb10f16a73ecc98da0b4dddfdf39278c35b1067afR276.

Reply:
Right. I was more asking, do we want the default behavior to be keep-alives, or the old behavior of disabling them by default (and closing connections)?